lib/pengine/utils.c

```diff
@@ -798,18 +797,18 @@ unpack_operation_on_fail(pe_action_t * action)
         for (operation = pcmk__xe_first_child(action->rsc->ops_xml);
              (operation != NULL) && (value == NULL);
              operation = pcmk__xe_next(operation)) {
+            bool enabled;
 
             if (!pcmk__str_eq((const char *)operation->name, "op", pcmk__str_none)) {
                 continue;
             }
             name = crm_element_value(operation, "name");
             role = crm_element_value(operation, "role");
             on_fail = crm_element_value(operation, XML_OP_ATTR_ON_FAIL);
-            enabled = crm_element_value(operation, "enabled");
             interval_spec = crm_element_value(operation, XML_LRM_ATTR_INTERVAL);
             if (!on_fail) {
                 continue;
-            } else if (enabled && !crm_is_true(enabled)) {
+            } else if (pcmk__xe_bool_attr(operation, "enabled", &enabled) == pcmk_rc_ok && !enabled) {
```

Comment:
technically this gives different behavior for the invalid format case, but enabled should default to true so it's an improvement